packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -140,10 +141,15 @@ export class SummaryManager implements IDisposable {
         // enforce connectedState.clientId === clientElection.electedClientId. But once we're Running, we should
         // only transition to Stopping when the electedParentId changes. Stopping the summarizer without
         // changing the electedParent will just cause us to transition to Starting again.
-        if (this.connectedState.clientId !== this.clientElection.electedParentId ||
-            (this.state !== SummaryManagerState.Running &&
-                this.connectedState.clientId !== this.clientElection.electedClientId)) {
-            return { shouldSummarize: false, stopReason: "parentShouldNotSummarize" };
+
+        // Conditions for returning notElectedParent/notElectedClient and not running the last summary:
+        // a) New Parent has been elected and it is not the current client, or
+        // b) We are not already running the summarizer and we are not the current elected client id.
+        if (this.connectedState.clientId !== this.clientElection.electedParentId) {
+            return { shouldSummarize: false, stopReason: "notElectedParent" };
+        } else if (this.state !== SummaryManagerState.Running &&
```

Comment:
#nit and unrelated to your change, but this would IMO look better without the `else`s.

```
if (this.connectedState.clientId !== this.clientElection.electedParentId) {
    return { shouldSummarize: false, stopReason: "notElectedParent" };
}

if (this.state !== SummaryManagerState.Running && 
          this.connectedState.clientId !== this.clientElection.electedClientId) {
    return { shouldSummarize: false, stopReason: "notElectedClient" };
}

if (!this.connectedState.connected) {
    return { shouldSummarize: false, stopReason: "parentNotConnected" };
}

(...)
```

Reply:
I can surely do it as I don't have a strong preference.